pkg/reconciler/instances/istio/actions/performer_test.go

```diff
@@ -292,6 +292,29 @@ func Test_DefaultIstioPerformer_PatchMutatingWebhook(t *testing.T) {
 		require.Contains(t, gotNew.Webhooks[0].NamespaceSelector.MatchExpressions, want)
 		require.NotContains(t, gotOld.Webhooks[0].NamespaceSelector.MatchExpressions, want)
 	})
+
+	t.Run("patching mutating webhook should be idempotent", func(t *testing.T) {
+		// given
+		whConfName := "istio-revision-tag-default"
+		kubeClient := mocks.Client{}
+		mutatingWebhookConf := createIstioMutatingWebhookConf(whConfName)
+		clientset := fake.NewSimpleClientset(mutatingWebhookConf)
+		kubeClient.On("Clientset").Return(clientset, nil)
+		wrapper := NewDefaultIstioPerformer(nil, nil, nil)
+
+		// when
+		err := wrapper.PatchMutatingWebhook(context.TODO(), &kubeClient, log)
+		require.NoError(t, err)
```

Comment:
From my point of view `when` section should contain at most one instruction. In some specific cases more. But any following assertion is part of `then` as we validate the state after performing an action.

What I really like is creating a transparent flow, sometimes combining many `when` and `then`. I think it is not a problem to add another `// and when` bound with `// and then` - it highly increases readability and underlines that there was once again a logic performed.